Update documentation (#63)
yiisoft/yii-queue: docs/guide/usage.md

Configuration
-------------

> In order to use the extension you can configure it with DI container like the following:

Reviewer: ```suggestion
In order to use the extension you can configure it with a DI container in the following way:
```

> In order to use the extension you can configure it with a DI container in the following way:

Reviewer 2: ```suggestion
You can configure it with a DI container in the following way:
```

```php
$eventDisptacher = $DIContainer->get(\Psr\EventDispatcher\EventDispatcherInterface::class);
$logger = $DIContainer->get(\Psr\Log\LoggerInterface::class);

$worker = $DIContainer->get(\Yiisoft\Yii\Queue\Worker\WorkerInterface::class);
$loop = $DIContainer->get(\Yiisoft\Yii\Queue\Cli\LoopInterface::class);
$driver = $DIContainer->get(\Yiisoft\Yii\Queue\Driver\DriverInterface::class);

$queue = new Queue(
    $driver,
    $eventDisptacher,
    $worker,
    $loop,
    $logger
);
```

> Documentation for drivers([synchronous driver](driver-sync.md), [AMQP driver](https://github.com/yiisoft/yii-queue-amqp)), 
> [workers](worker.md)

Reviewer: ```suggestion
See also the documentation for concrete drivers ([synchronous driver](driver-sync.md), [AMQP driver](https://github.com/yiisoft/yii-queue-amqp)) and [workers](worker.md)
```

For example, if you need to download and save a file the class may look like the following:

```php
class DownloadJob implements Yiisoft\Yii\Queue\Payload\PayloadInterface
{
    public $url;
    public $file;
    
    public function __construct(string $url, string $file)
    {
        $this->url = $url;
        $this->file = $file;
    }
    
    public function getName(): string
    {
        return 'earlyDefinedQueueHandlerName';
    }

    public function getData()
    {
        return function () {
            file_put_contents($this->file, file_get_contents($this->url));
        };

>     }

Reviewer: Here must be serializable data, not message handler implementation.

Job status
----------

```php
// Push a job into the queue and get a message ID.
$id = $queue->push(new SomeJob());

> //Get status of job

Reviewer: ```suggestion
// Get job status
```

$status = $queue->status($id);

// Check whether the job is waiting for execution.
$status->isWaiting();

// Check whether a worker got the job from the queue and executes it.
$status->isReserved($id);

// Check whether a worker has executed the job.
$status->isDone($id);
```

Handling events
---------------

The queue triggers the following events:

| Event class        | Triggered                                                 |
|--------------------|-----------------------------------------------------------|
| BeforePush         | before adding a job to queue using `Queue::push()` method |
| AfterPush          | after adding a job to queue using `Queue::push()` method  |
| BeforeExecution    | before executing a job                                    |
| AfterExecution     | after successful job execution                            |
| JobFailure         | on uncaught exception during the job execution            |

Logging events
--------------

> In order to log events, please refer to documentation of implementation of EventDispatcherInterface

Reviewer: ```suggestion
In order to log events, please refer to EventDispatcherInterface implementation documentation
```

processes. Therefore avoid external dependencies when executing a task if you're not sure if they are available in
the environment where the worker does its job.

> All the data to process the task should be provided with data of your payload

Reviewer: ```suggestion
All the data to process the task should be provided with your payload `getData()` method.
```
